Support Django 5.2.1 and elasticsearch-dsl 7.x

ddusim opened the pull request:
- Update setup.py dependencies to elasticsearch-dsl>=7.0.0,<8.0.0
- Add Django 5.x classifiers and Python 3.12 support
- Fix DocType import (replaced with Document for elasticsearch-dsl 7.x)
- Bump version to 7.0.0a1

ddusim committed:
Support Django 5.2.1 and elasticsearch-dsl 7.x

- Update setup.py dependencies to elasticsearch-dsl>=7.0.0,<8.0.0
- Add Django 5.x classifiers and Python 3.12 support
- Fix DocType import (replaced with Document for elasticsearch-dsl 7.x)
- Bump version to 7.0.0a1

kofic commented:
@ddusim I don't think we should merge to master? Can we tag this branch and have CP requirements install that tag?

I believe that's what the previousl upgrades did - I don't believe they were merged to master

ddusim committed:
Fix elasticsearch-dsl compatibility: Remove deprecated super().doc_type() calls

- Replace super().doc_type() calls with direct doc_type handling
- Fixes AttributeError: 'super' object has no attribute 'doc_type'
- Compatible with newer elasticsearch-dsl versions that removed doc_type method
- Maintains backward compatibility by using stored __doc_type attribute

ddusim commented:
@kofic I am using this in requirements.in git+https://github.com/HBS-HBX/django-elastic-migrations.git@django-5.2.1-elasticsearch-7-clean and I don't think we need to merge this branch

kofic commented:
@ddusim nice looks great 💯
